ddp-workspace/projects/ddp-dsm-ui/src/app/phi-manifest/phi-manifest.component.ts

```diff
@@ -0,0 +1,112 @@
+import {Component, OnInit} from '@angular/core';
+import {Auth} from '../services/auth.service';
+import {ComponentService} from '../services/component.service';
+import {DSMService} from '../services/dsm.service';
+import {Statics} from '../utils/statics';
+import {Utils} from '../utils/utils';
+import {ActivatedRoute} from '@angular/router';
+import {PhiManifestResponseModel} from './phi-manifest-response.model';
+import {PhiManifestModel} from './phi-manifest.model';
+
+@Component({
+    selector: 'app-phi-manifest',
+    templateUrl: './phi-manifest.component.html',
+    styleUrls: ['./phi-manifest.component.scss']
+})
+
+export class PhiManifestComponent implements OnInit {
+    errorMessage: string;
+    additionalMessage: string;
+    loading = false;
+    realm: string;
+    participantId: string = null;
+    sequencingOrderId: string = null;
+    allowedToSeeInformation = false;
+
+    constructor(private dsmService: DSMService, private auth: Auth, private route: ActivatedRoute) {
+        if (!auth.authenticated()) {
+            auth.sessionLogout();
+        }
+        this.route.queryParams.subscribe(params => {
+            this.realm = params[DSMService.REALM] || null;
+            if (this.realm != null) {
+                this.checkRight();
+            }
+        });
+    }
+
+    private checkRight(): void {
```

Comment:
What's being checked here?  That this is to the right of the page?  The method name should be descriptive.

Reply:
The checkRight methods are a pattern across all of the components in the DSM frontend that check if the user has rights to view this component for this study